Changes for new Droid and backend checks

MancunianSam opened the pull request:
* Move the lambda to use the new Droid library
* Remove the SQS queues

TomJKing commented on project/Dependencies.scala:
```diff
@@ -2,26 +2,26 @@ import sbt._
+  lazy val apacheCommons = "org.apache.commons" % "commons-lang3" % "3.12.0"
+  lazy val droidApi = "uk.gov.nationalarchives" % "droid-api" % "6.6.0-rc2"
+  // This is an older version of this dependency but the newer version won't work with Droid without some major changes.
+  // I'll configure Scala Steward to ignore it.
```
Maybe change this second comment to: `Scala Steward configured to ignore it`

TomJKing commented on project/Dependencies.scala:
```diff
@@ -2,26 +2,26 @@ import sbt._
   lazy val wiremock = "com.github.tomakehurst" % "wiremock" % "2.27.2"
+  lazy val apacheCommons = "org.apache.commons" % "commons-lang3" % "3.12.0"
+  lazy val droidApi = "uk.gov.nationalarchives" % "droid-api" % "6.6.0-rc2"
+  // This is an older version of this dependency but the newer version won't work with Droid without some major changes.
```
Is there anything on the `DROID` backlog about updating this?

MancunianSam replied:
I'm not sure, from the brief look I had, it's a large job so it won't be any time soon. They're getting a contractor in to look at Droid soon though so maybe then.

TomJKing replied:
Fair enough, was just wondering if there was a task we could reference about this.

TomJKing commented on src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Missing line break at end of file.

TomJKing replied:
Missing line break at end of file.

TomJKing reviewed (commented):
Added some minor comments.

I tried to run locally getting an error when trying to download the DROID snapshot version?
```
[error] sbt.librarymanagement.ResolveException: Error downloading uk.gov.nationalarchives:droid-api:6.6.0-SNAPSHOT
[error]   Not found

```

Some other general comments:

- should update the README to take account of the new architecture
- there are a couple of `Jenkins` files which I think can be deleted from the repo now

MancunianSam commented:
I'm not sure why you're getting not found on the snapshot version. Can you try clearing out the target directory and trying again, it sounds like something weird is cached somewhere.

I've removed the Jenkinsfiles. I've also removed the docker image which writes Droid to EFS, the job which deploys the docker image and the job that runs it. 

I've changed the README as well.

TomJKing commented on src/main/scala/uk/gov/nationalarchives/fileformat/LambdaRunner.scala:
```diff
@@ -0,0 +1,17 @@
+import scala.jdk.CollectionConverters._
+
+object LambdaRunner extends App {
+    val event = new SQSEvent()
```
Is this `val event` used anywhere?

MancunianSam replied:
No, it's not, I've removed it.

TomJKing commented on src/main/scala/uk/gov/nationalarchives/fileformat/LambdaRunner.scala:
```diff
@@ -0,0 +1,17 @@
+package uk.gov.nationalarchives.fileformat
+
+import com.amazonaws.services.lambda.runtime.events.SQSEvent
+import com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage
```
Not sure this import is needed? `import com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage`

MancunianSam replied:
It's not needed and the entire library isn't needed. I've removed this and the AWS library they're imported from.

TomJKing commented on src/test/scala/uk/gov/nationalarchives/fileformat/FileSpec.scala:
```diff
@@ -0,0 +1,49 @@
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach, Suite}
+import com.github.tomakehurst.wiremock.WireMockServer
```
Duplicate import with line 3: `import com.github.tomakehurst.wiremock.WireMockServer`

TomJKing commented on src/test/scala/uk/gov/nationalarchives/fileformat/FileSpec.scala:
```diff
@@ -0,0 +1,49 @@
+    super.beforeAll()
+  }
+
+  // Create the files and then try to ls them in the test script. This mimics reading from the existing file on EFS
```
I thought we no longer need EFS? Maybe this comment needs updating?

TomJKing commented on src/test/scala/uk/gov/nationalarchives/fileformat/LambdaTest.scala:
```diff
@@ -0,0 +1,145 @@
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers.{equal, _}
+import org.scalatest.prop.{TableDrivenPropertyChecks, TableFor1, TableFor2}
```
Unused import: `import org.scalatest.prop.{..., TableFor1, ...}`

TomJKing commented on README.md:
```diff
@@ -4,30 +4,24 @@ This is the repository for the backend file format check for the [Transfer Digit
+Droid is contained within the `"uk.gov.nationalarchives" % "droid-api"` dependency. 
+The binary and container signature files are downloaded when the lambda is first started. 
+The versions of the signature files are stored in the `DROID_SIGNATURE_VERSION` and `CONTAINER_SIGNATURE_VERSION` environment variables. 
+To update to a new version of these signature files, we need to update the values in the environment variables.
```
Is this a manual process to find out if there is a new version? Probably should note where to find the latest versions.

MancunianSam replied:
It is a manual process at the moment. I'm hoping there will be some kind of message we can subscribe to to automatically update them but there's nothing yet. I've put this in the README along with where to find the signature files.

TomJKing reviewed (commented):
Some additional comments. Clearing out the cache sorted out downloading the `DROID` snapshot version.

TomJKing reviewed (approved):
Looks good. Thanks
